README.md

```diff
@@ -275,6 +275,126 @@ An example policy would look like;
 }
 ```
 
+## Message formats and AWS interactions
+
+JustSaying uses a number of Amazon Web Services APIs to transport messages between publishers and subscribers. Since these HTTP APIs can be used by applications which do not use JustSaying, it is entirely possible to have, for example, a Java application publishing a message which is subscribed to by an application using JustSaying. Equally, a C# application publishing messages using JustSaying, which are subscribed to by a Java application, is fully supported by AWS. In order to support this interoperability, it is important that the actual message formats are described. Since Amazon provide SDKs for many programming languages and frameworks, you are unlikely to interact with the APIs purely via HTTP, but knowing the structure of the JustSaying message JSON itself is useful for cross-language purposes.
+
+### Publishing to SNS
+
+JustSaying uses the SNS Publish API to send messages to SNS (documented [here](https://docs.aws.amazon.com/sns/latest/api/API_Publish.html)). The following parameters are set:
+
+#### TopicArn
+The ARN of the topic to which the message is being published.
+
+#### Subject
+JustSaying will set this to the type of the .NET message class - "OrderAccepted", for example.
+
+#### Message
+The actual message object, serialised to JSON. Because JustSaying messages must derive from the base Message class, the properties on that base class will be serialised into this JSON. On the base class, the `Id` property is set as a random GUID and the `TimeStamp` property is set to the current time (in UTC). The default serialisation settings will ignore null properties, so properties like `Conversation` will not appear in the JSON if they are left as the default null value. Enums are serialised as strings.
+
+In summary, suppose we have the following message class:
+
+````c#
+        public class OrderAccepted : Message
+        {
+            public OrderAccepted(int orderId)
+            {
+                OrderId = orderId;
+            }
+            public int OrderId { get; private set; }
+        }
+````
+
+And suppose I create a message object from that class:
+
+````c#
+        var orderAccepted = new OrderAccepted(1234)
+        {
+            RaisingComponent = "my publisher",
```

Comment:
Is there any guidance around what `RaisingComponent` should actually be? `my publisher` is confuses me.

Reply:
Internally it's generally the name of a feature/application, such as `"orderapi"`.

Reply:
Nope. Internally at Just Eat, we set it to the name of the app (the independently-deployable component) that's publishing the message.

Reply:
Given the confusion already it'd be nice to see a recommendation

Reply:
I added a table that lists the intended use of these fields (or at least what I assume it to be based on the name!) Also tidied up the examples to use "orderapi" for `RaisingComponent` which should make things clearer.